### PR TITLE
Orderbook updates

### PR DESCRIPTION
**Pull request by ddervs**
1. Speeds up transacted volume computation
  2. Speeds up conversion of log to DataFrame format
  3. Adds `wide-book` compatibility for `book-freq` != 0
  4. `Order._order_ids` collection of already-used order IDs changed from `list` to `set` for 
efficiency.

### CODE REVIEW

File: util/OrderBook.py [hunk 1]
```diff
@@ -452,6 +493,42 @@ def isEqualPrice(self, order, o):
...
+        :return:
+        """
+        quotes = sorted(list(self.quotes_seen))
+        log_len = len(self.book_log)
+        quote_idx_dict = {quote: idx for idx, quote in enumerate(quotes)}
+        quotes_times = []
+
+
+        # Construct sparse matrix, where rows are timesteps, columns are quotes and elements are volume.
+        S = dok_matrix((log_len, len(quotes)), dtype=int)  # Dictionary Of Keys based sparse matrix.
```
Comment (mamahfouz) [inline util/OrderBook.py:518]: very nice!!

File: util/order/Order.py [hunk 2]
```diff
@@ -1,36 +1,47 @@
...
   _order_ids = []
 
   def __init__(self, agent_id, time_placed, symbol, quantity, is_buy_order, order_id=None, tag=None):
+
     self.agent_id = agent_id
 
     # Time at which the order was created by the agent.
-    self.time_placed = time_placed
+    self.time_placed: pd_Timestamp = time_placed
+    # print(f'time placed: {self.time_placed}')
```
Comment (mamahfouz) [inline util/order/Order.py:19]: just remove that print line but not a big issue

File: util/OrderBook.py (hunk 1 above)
Comment (ddervs) [reply util/OrderBook.py:518]: Thanks!

File: util/order/Order.py (hunk 2 above)
Comment (ddervs) [reply util/order/Order.py]: Good spot! Done

**Pull request merged**